pac4j-cas/src/main/java/org/pac4j/cas/client/rest/CasRestClient.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright 2012 - 2015 pac4j organization
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.pac4j.cas.client.rest;
+
+
+import org.apache.commons.httpclient.HttpStatus;
+import org.jasig.cas.client.authentication.AttributePrincipal;
+import org.jasig.cas.client.validation.Assertion;
+import org.jasig.cas.client.validation.TicketValidationException;
+import org.pac4j.cas.credentials.CasCredentials;
+import org.pac4j.cas.profile.CasProfile;
+import org.pac4j.core.client.BaseClient;
+import org.pac4j.core.client.Mechanism;
+import org.pac4j.core.context.WebContext;
+import org.pac4j.core.exception.TechnicalException;
+import org.pac4j.http.client.direct.DirectHttpClient;
+import org.pac4j.http.credentials.UsernamePasswordCredentials;
+import org.pac4j.http.credentials.extractor.Extractor;
+import org.pac4j.http.credentials.extractor.FormExtractor;
+import org.pac4j.http.profile.HttpProfile;
+import org.pac4j.http.profile.creator.AuthenticatorProfileCreator;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.OutputStreamWriter;
+import java.net.HttpURLConnection;
+import java.net.URL;
+
+/**
+ * This is {@link CasRestClient}.
+ *
+ * @author Misagh Moayyed
+ * @author Daniel Pacak
+ * @since 1.8.0
+ */
+public class CasRestClient extends DirectHttpClient<UsernamePasswordCredentials> {
```

Comment:
Got it. I do think this is just going to lead to API proliferation because at this point, the mechanism field is more of a nuisance perhaps but as you said in the other comment, it might be better to hide the extractor directly inside a dedicated class.